### PR TITLE
GitHub Actions workflow for black code formatting

### LINKED ISSUE #8: [Other] Code formatting using `black` (and more)
It does currently not seem like paperless-ng(x) is following any particular coding style.

In order to keep everything uniform and prevent certain kinds of discussion, I'd suggest formatting all Python code with [black](https://github.com/psf/black).
As a next step, one could format the TypeScript code using, e.g. [prettier](https://prettier.io/). And linting with, e.g. [flake8](https://github.com/PyCQA/flake8) might also be a good idea.

What do you guys think? I'd be happy to create a PR and then also add code formatting tests to our GitHub Actions workflows.

### PR DESCRIPTION
**Pull request by kpj**
This PR adds a GitHub Actions workflow which uses [black](https://github.com/psf/black) to check for Python code formatting issues.
It also removes the calls to `check-trailing-newline` and `check-trailing-whitespace` which are now no longer needed.
Closes #8.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -80,21 +80,19 @@ jobs:
         name: Codestyle
         run: |
           cd src/
-          pycodestyle
-  whitespace:
+          pycodestyle --max-line-length=88 --ignore=E121,E123,E126,E226,E24,E704,W503,W504,E203
```
Comment (bauerj) [inline .github/workflows/ci.yml:83]: Do we still need this?

**Comment by tribut** [comment]
Should we keep check-trailing-newline and check-trailing-whitespace for non-python files?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kpj) [reply .github/workflows/ci.yml:83]: While there is certainly a substantial overlap between the things black and pycodestyle check, pycodestyle still has a few additions, which is why I didn't delete it right away.
They include, e.g., "‘async’ and ‘await’ are reserved keywords starting with Python 3.7" ([full list](https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes)).

I'd be in favor of replacing it with [flake8](https://github.com/PyCQA/flake8) which is a wrapper for various other tools or removing it completely.

**Comment by kpj** [comment]
@tribut For most other files (TypeScript, Bash, Markdown) I'd rather use language-specific linters/formatters, but for all other files this might indeed be a good idea.
I can readd these two rules if you want :-)

**Comment by bauerj** [comment]
I don't really see much value added by those two rules. I'm not strongly opposed to it but I'd say we keep it simple.

**Comment by bauerj** [comment]
You should probably also add a note about black and how best to use it to [CONTRIBUTING.md](https://github.com/paperless-ngx/paperless-ngx/blob/dev/CONTRIBUTING.md) and/or [the docs](https://paperless-ng.readthedocs.io/en/latest/extending.html).

**Comment by qcasey** [comment]
The rules would apply notably to the `.rst` text for the docs. Although I also don't see value in removing trailing whitespace. Let's leave them out for now.

Should black go into requirements.txt?

**Comment by bauerj** [comment]
> Should black go into requirements.txt?

No, but into `Pipfile` under `dev-packages`.

**Comment by kpj** [comment]
I added black to `Pipfile`, but updating `Pipfile.lock` lead to build errors on my machine, so I reverted the lock update.

**Comment by stumpylog** [comment]
> You should probably also add a note about black and how best to use it to [CONTRIBUTING.md](https://github.com/paperless-ngx/paperless-ngx/blob/dev/CONTRIBUTING.md) and/or [the docs](https://paperless-ng.readthedocs.io/en/latest/extending.html).

One other option is added a `pre-commit` configuration file, with a hook to run `black`.  That would help contributors to have files already formatted.  This would also want `pre-commit` in the dev-packages section of the Pipfile

**Comment by bauerj** [comment]
Yes, but I think that should go in a separate PR.

**Review by qcasey — APPROVED** [review]

**Review by bauerj — APPROVED** [review]
Excellent

**Pull request merged**